apps/studio/data/database/index-advisor-enable-mutation.ts

```diff
@@ -0,0 +1,241 @@
+import { useMutation, UseMutationOptions } from '@tanstack/react-query'
+import { toast } from 'react-hot-toast'
+
+import { executeSql } from 'data/sql/execute-sql-query'
+import type { ResponseError } from 'types'
+
+// [Joshen] This is just temporary while we wait for Index Advisor to be baked into the image itself
+
+export type IndexAdvisorEnableVariables = {
+  projectRef: string
+  connectionString?: string
+}
+
+export async function enableIndexAdvisor({
+  projectRef,
+  connectionString,
+}: IndexAdvisorEnableVariables) {
+  const sql = `
+create or replace function index_advisor(
```

Comment:
Is this function used in this PR? Also creating this function in the `public` schema is probably not the best idea since it can be called from postgrest

Reply:
so this is just temporary until the index_advisor gets to staging/prod as a database extension - we'd remove this mutation at the end ideally

Reply:
but yes will be used in the subsequent PR when we integrate the index advisor stuffs